source/extensions/common/wasm/context.cc

```diff
@@ -1589,7 +1589,14 @@ void Context::setEncoderFilterCallbacks(Envoy::Http::StreamEncoderFilterCallback
   encoder_callbacks_ = &callbacks;
 }
 
-void Context::onHttpCallSuccess(uint32_t token, Envoy::Http::ResponseMessagePtr& response) {
+void Context::onHttpCallSuccess(uint32_t token, Envoy::Http::ResponseMessagePtr&& response) {
+  if (proxy_wasm::current_context_ != nullptr) {
+    // We are in a reentrant call, so defer.
+    auto response_ptr = response.release();
+    wasm()->addAfterVmCallAction([this, token, response_ptr] {
+      onHttpCallSuccess(token, std::unique_ptr<Envoy::Http::ResponseMessage>(response_ptr));
+    });
+  }
```

Comment:
Shouldn't there be a `return;` at the end of each `if` statement? Otherwise, you're going to execute both code patches.

Reply:
Woops.  Done.